compiler/qsc_frontend/src/typeck/convert.rs

```diff
@@ -146,6 +148,11 @@ pub(super) fn ast_ty_def(names: &Names, def: &TyDef) -> (UdtDef, Vec<MissingTyEr
                     })
                     .collect(),
             ),
+            TyDefKind::Err => UdtDefKind::Field(UdtField {
+                name_span: None,
+                name: None,
+                ty: Ty::Err,
+            }),
```

Comment:
I have no idea how this code is invoked, but this one makes me nervous because the handling looks nontrivial compared to the other `Err` cases. Can we have a unit test case that exercises this specifically?

Also, should we be pushing an error into `errors` when we encounter this node? Or is the idea that the error should have been previously reported for this node anyway?

Reply:
I can add a test for this one to confirm my understanding. And yes, the expectation is that error entries in the AST/HIR should already correspond to errors that were filed during parsing, so there is no need to file an error here. This spot just needs to be able to handle an error in the type definition without failing to produce a type so that we can eventually get the right HIR representing the partially defined type.

Reply:
Actually, try as I might, I cannot cause the code to go down this path. I will turn this into a panic rather than leave unused branch in here.

Reply:
This should be reachable with `newtype Bar = (, y : Int);`.

Reply:
I just confirmed that `newtype Bar = (, y : Int);` throws the added panic, and in the commit before the panic was added, it gave the expected behavior of yielding `newtype Bar = (?, y : Int)`.

Reply:
Ah, you're right. I see where my mistake was: I was counting on the tests in the parser which already have a case for this, but they just show AST. This code path produces a `qsc_hir::ty::UdtDef` so it only gets invoked when we are starting to calculate the relevant HIR structures. What this tells me is that I'm missing tests on the HIR error kinds that we expect to see, so I'll add those cases to lower/tests.rs as part of putting back this logic. Good catch Scott!